lib/msf/core/exploit/remote/http/nagios_xi/login.rb

```diff
@@ -31,11 +27,16 @@ def nagios_xi_login(user, pass)
       return Msf::Exploit::CheckCode::Unknown('Unable to obtain the value of the `nsp_str` token required for authentication')
     end
 
+    pre_auth_cookies = res_pre_login.get_cookies
+    if pre_auth_cookies.blank?
+      return Msf::Exploit::CheckCode::Unknown('Unable to obtain the cookies required for authentication')
```

Comment:
Presumably this is due to authentication failure, in which case `Failure::NoAccess` is more accurate.